item2/Dictionary.java

```diff
@@ -0,0 +1,53 @@
+import java.util.*;
+
+public class Dictionary {
+  private Map<String, Object> dict = new HashMap<String, Object>();
+
+  private Dictionary(Builder builder) {
+    dict = builder.dict;
+  }
+
+  @Override
+  public String toString() {
+    String str = "{\n";
+    for (String key : dict.keySet()) {
+      String[] valueStrLines = dict.get(key).toString().split("\\r?\\n");
+      str += "\t\"" + key + "\" : " + valueStrLines[0] + "\n";
+      for (int i = 1; i < valueStrLines.length; i++)
+        str += "\t" + valueStrLines[i] + "\n";
+    }
+    str += ("}");
+    return str;
+  }
+
+  public static Builder createObjectBuilder() {
+    return new Builder();
+  }
+
+  public void printDictionary() {
+    System.out.println(this);
+  }
+
+  public static class Builder {
```

Comment:
Since we are accessing it using Dictionary.Builder I believe this naming convention looks more accurate than Dictionay.DictionaryBuilder. DictionaryBuilder would have been appropriate if we are creating object using something like DictionaryBuilder db = new DictionaryBuilder(). Please let me know if you still want me to change it.